src/command-helpers/network.js

```diff
@@ -0,0 +1,135 @@
+const path = require('path')
+const yaml = require('yaml')
+const { step, withSpinner } = require('../command-helpers/spinner')
+
+const updateSubgraphNetwork = async (toolbox, manifest, network, networksFile) =>
+  await withSpinner(
+    `Update sources network`,
+    `Failed to update sources network`,
+    `Warnings while updating sources network`,
+    async spinner => {
+      let allNetworks
+
+      try {
+        step(spinner, `Reading networks config`)
+        allNetworks = await toolbox.filesystem.read(networksFile, "json")
+      }  catch (error) {
+        throw new Error(error.message)
+      }
+
+      let networkConfig = allNetworks[network]
+
+      // Exit if the network passed with --network does not exits in networks.json
+      if(!networkConfig) {
+        throw new Error(`Network '${network}' was not found in '${networksFile}'`)
+      }
+
+      await toolbox.patching.update(manifest, content => {
+        let subgraph = yaml.parse(content)
+        let networkSources = Object.keys(networkConfig)
+        let subgraphSources = subgraph.dataSources.map(value => value.name);
+
+        // Update the dataSources network config
+        subgraph.dataSources = updateSources(
+          toolbox,
+          spinner,
+          network,
+          subgraph.dataSources,
+          networkSources,
+          networkConfig
+        )
+
+        // All data sources shoud be on the same network,
+        // so we have to update the network of all templates too.
+        if(subgraph.templates) {
+          subgraph.templates = updateTemplates(network, subgraph.templates)
```

Comment:
Personal opinion, but I don't think you need `updateTemplates` function.

If you prefer to keep it, I would use `.map` instead of `.forEach` since you're already reasigning the value here.

Here's a way to do it, but you can do a regular `if` statement if you think it's more readable:

```javascript
subgraph.templates = subgraph.templates.map(template => ({
  ...template,
  network: template.network !== network
    ? network
    : template.network,
}))
```

In fact I don't think you even have to do the check since it will give you the same result, the Node.js engine might optimize this already. Either way I don't think this will be a performance bottleneck.

```javascript
subgraph.templates = subgraph.templates.map(template => ({
  ...template,
  network,
}))
```